### PR TITLE
Add environment leak test to bloat test CI

### PR DESCRIPTION
**Pull request by jentfoo**
This is a new test which exports an environment variable, and then validates that none of the build binaries contain the unexpected variable. This is designed to help make sure that secrets don't unexpectedly leak from any of our build environments.

This PR is designed to be a low hanging fruit attempt at resolution for this backlog improvement: https://github.com/gravitational/SecOps/issues/340

### CODE REVIEW

**Review by zmb3 — COMMENTED** [review]
Hmm, is it worth running this check on every single PR for something that happened once in several years?

**Comment by jentfoo** [comment]
@zmb3 I think it's worth some sort of check, but maybe not this exact design.  I view the recent issue as a close miss, we got too lucky to want to dance with it every few years.  So I think some type of test makes sense to try and guard against this risk.

That said, I agree it's unfortunate to run ~5min of actions time on every PR.  Let me know if you have any thoughts on alternative ways for doing this check.

**Comment by zmb3** [comment]
Right. This was a result of changing the build system, which is not a frequent thing. (To my knowledge this has been done exactly once in teleport's 8 year history)

Running a check on each PR (dozens of times per day) to protect against something that changes less than once a year seems wasteful. 

That said, we are planning on adding a binary size check where we will have to build binaries anyway. I wasn't planning on including teleport and it's web UI in scope, but we could include this test there. 

Alternatively, maybe we only run this as part of the release pipeline?

**Comment by jentfoo** [comment]
> Alternatively, maybe we only run this as part of the release pipeline?

I considered this.  However when dev credentials are considered it would be detecting the issue too late.  This would be better than nothing, but is not a complete solution.  Still, maybe it's the best balance?

> That said, we are planning on adding a binary size check where we will have to build binaries anyway. I wasn't planning on including teleport and it's web UI in scope, but we could include this test there.

When were you planning on doing this check?  I think combining this with other binary tests makes total sense.  I just did not see any test where I could piggy back on fully compiled binaries.

**Review by wadells — COMMENTED** [review]
IMO it too costly to run an entire build for this check.  If we could integrate it into one of the jobs where we're already building, we'd capture most of the value for 

I think doing the check at tag build time is fine.  This means developers may leak their environments, but we won't leak any release infra environments.  Thus I'd suggest https://github.com/gravitational/teleport/blob/36af31d124d8924fbafef4530ff754b3790a09a6/.github/workflows/build-centos7-assets.yaml as a potential target.

**Comment by jentfoo** [comment]
I have updated this PR so that the test only occurs at the merge queue.  Since this is assumed to be unlikely this seems like a decent compromise between costs and verifying there was not an environment leak.

I also took a shot at estimating these cost....
April averaged 16 `master` merges per day (448 total)
So far in May we have averaged 17 `master` merges per day (up to May 15th, 247 total)

So for estimating I am assuming a given merge queue job will run on average 20 times per day.  With 20 builds a day we are assuming that 25% of merge queue jobs need to be retried.

Given those build rates this test would cost us an estimated $180 a month.

I think ultimately this would be best to combine with the work @zmb3 mentioned above.  But until then I think we should land this PR as an initial step to protect against environment secrets leaking.  To me the potential impact of such an exposure is worth being proactive.

@zmb3, @wadells what do you think?  Even if this is not perfect, does the cost seem proportional to the benefit?

**Comment by jentfoo** [comment]
@zmb3 Can you review this now?  We have combined the environment leak test as part of our `bloat` test

File: .github/workflows/bloat.yaml
Comment (zmb3) [inline .github/workflows/bloat.yaml:68]:
```suggestion
          BUILD_SECRET=FAKE_SECRET make BUILDDIR=base_build ensure-webassets binaries
```

File: Makefile
Comment (zmb3) [inline Makefile:703]: I don't see this being called anywhere.

Comment (jentfoo) [reply Makefile:703]: The workflow could execute this, but the failure cause may be less obvious.  So we did not use this directly in the workflow.

I wanted to keep this though.  I think it's important that test can be run locally

File: .github/workflows/bloat.yaml
Comment (jentfoo) [reply .github/workflows/bloat.yaml:68]: `make full` sets this explicitly, I felt like doing the same here helped make it explicit that we also want to include the webassets as part of this test.

File: .github/workflows/bloat.yaml
Comment (jentfoo) [reply .github/workflows/bloat.yaml]: Thank you, I fixed it in https://github.com/gravitational/teleport/pull/26033/commits/b5f3ae4732cb134b105a7f3c421f66d939276909

File: .github/workflows/bloat.yaml
Comment (jentfoo) [reply .github/workflows/bloat.yaml:22]: This is correct.  Personally I think we should just let either failure prevent the other from completing.  It's much simpler, and an environment leak failure will be very rare.

Comment (jentfoo) [reply .github/workflows/bloat.yaml:22]: Considering how rare the env failure should be, this is not a huge concern for me.  If you want I can adopt your changes, but I get concerned with more complex workflows that are not frequently verified anyways (ie not likely to have a failure)

Let me know what you think, thank you!

File: Makefile
Comment (zmb3) [reply Makefile:703]: Ok, let's call it `test-env-leakage` or something then. `test-env` sounds like some sort of generic "prepare a test environment" step.

and remove the `called by CI/CD` comment since it is not true :-)

Comment (jentfoo) [reply Makefile:703]: Done, thank you!

**Comment by jentfoo** [comment]
@wadells @zmb3 @rosstimothy Can I get a review on this PR?  I believe this should be good to merge now.

File: Makefile
Comment (zmb3) [reply Makefile]: Mind updating the PR title too (so we don't get an inaccurate commit message)?

File: Makefile
Comment (zmb3) [inline Makefile:719]: Use tabs here?

**Review by zmb3 — APPROVED** [review]
LGTM if @rosstimothy is good with it

**Review by rosstimothy — APPROVED** [review]

File: .github/workflows/bloat.yaml
Comment (jentfoo) [reply .github/workflows/bloat.yaml]: I agree, I have updated this to just fail normally.  Let me know if you have any additional concerns.

**Pull request merged**
